libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1737,7 +1737,14 @@ void IRGeneratorForStatements::endVisit(MemberAccess const& _memberAccess)
 		}
 		else if (member == "name")
 		{
-			solUnimplementedAssert(false, "");
+			TypePointer arg = dynamic_cast<MagicType const&>(*_memberAccess.expression().annotation().type).typeArgument();
+			ContractDefinition const& contract = dynamic_cast<ContractType const&>(*arg).contractDefinition();
+			m_code << Whiskers(R"(
```

Comment:
Please use some kind of `define` helper function.

Reply:
Fixing